### DISCUSSION TITLE
Running roadrunner with no rr serve command

### DISCUSSION THREAD

**Discussion by rookiextreme** [roadrunner-server/roadrunner#1983] [Q&A] [answered]
Hi, new to roadrunner

ive been reading the documentation and a little confused on how is it to be run in nginx. Not using docker.

Do i need to run rr serve to any project using roadrunner? Because looking at the installation documentation, you need to specify the port of roadrunner, but it doesnt state how to run roadrunner rr serve(through configuration files, not command)

sorry if wrong information..

**Comment by rustatian** [comment, marked as answer]
Hey @rookiextreme 👋 
RR is a separate tool. NGINX is a web-server, RR is an application server. You may replace nginx with RR but in basic cases, generally it is not a web-server replacement.

**Reply by rookiextreme** [reply]
I see..ok because the whole concept is really new. If were to deploy how do we exactly automate the serve? Set a supervisor conf..?

**Reply by rustatian** [reply]
RR is a binary. You only need a `.rr.yaml` with a configuration. You may use `systemd` unit file (example can be found at docs.roadrunner.dev) or whatever daemonization tool you use.

**Reply by rookiextreme** [reply]
Alright, will do that, will update here if anything..thank you so much

**Reply by rookiextreme** [reply]
Because i want to develop in macos for now..for learning purposes

**Reply by rustatian** [reply]
RR supported on all platforms

**Reply by rookiextreme** [reply]
> RR supported on all platforms

Is there any reference or guide in the roadrunner to create such daemon in ubuntu/macos?

**Reply by rustatian** [reply]
docs.roadrunner.dev

**Reply by rookiextreme** [reply]
Yeap i have read the documentation for the systemd section.

Which means that for roadrunner, for development or even deployment, need to run rr serve manually OR daemonize it, am i right?

**Reply by rustatian** [reply]
RR is a binary, you can run it via docker/manually/automatically. Whatever applied to any binary, applied to RR.

**Reply by rookiextreme** [reply]
> RR is a binary, you can run it via docker/manually/automatically. Whatever applied to any binary, applied to RR.

Yeap, automatic you mean by systemd correct?

**Reply by rustatian** [reply]
As one of the options

**Reply by rookiextreme** [reply]
> As one of the options
Yeap, i have found the solution, using supervisor..so got supervisor in mac and in dev server and all is working now
